applications/compareSummary.cpp

```diff
@@ -0,0 +1,460 @@
+    ref_data_vec(nullptr),
+    checking_vec(nullptr),
+    check_data_vec(nullptr),
+    relative_tolerance_max(relative_tolerance_max),
+    relative_tolerance_median_max(relative_tolerance_median_max),
+    keys1(stringlist_alloc_new()),
+    keys2(stringlist_alloc_new()),
+    keys_short(stringlist_alloc_new()),
+    keys_long(stringlist_alloc_new())
+  {} 
+
+
+  bool open(const char* smspecFile1, const stringlist_type* unsmryFile1, const char* smspecFile2, const stringlist_type* unsmryFile2);
+  void close();
+  void setDataSets(std::vector<double> &time_vec1,std::vector<double> &time_vec2);
+  void setKeys(); // Creates a keylist containing all the keywords in a summary file. When the number of keywords are not
+  // equal in the two files of interest, it figures out which that contains most/less keywords
+  void setToleranceLevels(double relative_tolerance_max, double relative_tolerance_median_max);
+  void setTimeVecs(std::vector<double> &time_vec1,std::vector<double> &time_vec2);
+  void setDataVecs(std::vector<double> &data_vec1,std::vector<double> &data_vec2, int index1, int index2); //Read the data from the two files into separate vectors. Not necessary the same amount of values, but the values correspond to the same interval in time. Thus possible to interpolate values. 
+  void getDeviations(); 
+  void chooseReferance(std::vector<double> &time_vec1,std::vector<double> &time_vec2,std::vector<double> &data_vec1,std::vector<double> &data_vec2);
+  void evaluateDeviations(std::vector<double> &absdev_vec, std::vector<double> &reldev_vec);
+
+  static Deviation calculateDeviations( double val1, double val2); static double max(std::vector<double>& vec);
+  static double median(std::vector<double>& vec);
+  static double linearPolation(double check_value, double check_value_prev, double time_array[3]);
+  static void findDeviations(std::vector<double>& absdev_vec,std::vector<double>& reldev_vec,std::vector<double> *referance_vec,std::vector<double>* checking_vec ,std::vector<double> *ref_data_vec,std::vector<double> *check_data_vec);
+  static double average(std::vector<double> &vec);
+};
+
+
+
+
+
+
+int main (int argc, char ** argv){
+  if(argc != 5){
+    printHelp();
+    return 0;
+  }
+ try
+   {
+  const char * smspecFile1 = argv[1];
+  const char * unsmryFile1 = argv[2];
+  const char * smspecFile2 = argv[3];
+  const char * unsmryFile2 = argv[4];
+  double relative_tolerance_max = 1 ;
+  double relative_tolerance_median_max = 0.1;
+  stringlist_type * file1 = stringlist_alloc_new();
+  stringlist_type * file2 = stringlist_alloc_new();
+  stringlist_append_copy(file1, unsmryFile1);
+  stringlist_append_copy(file2, unsmryFile2);
+ 
+
+  SummaryReader read(relative_tolerance_max,relative_tolerance_median_max); //may also use sat function for tolerance limits
+  read.open(smspecFile1, file1, smspecFile2, file2);
+  read.setKeys();
+  read.getDeviations();
+  read.close();
+  stringlist_free(file1);
+  stringlist_free(file2);
+   }
```

Comment:
this was also a strange indentation.  Could you through the entire file and fix indentation issues?